microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/keycloak/KeycloakIntegrationHighLevelScenarioTest.java

```diff
@@ -0,0 +1,112 @@
+package org.jboss.eap.qe.microprofile.jwt.keycloak;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.eap.qe.microprofile.jwt.testapp.Endpoints;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.JaxRsTestApplication;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.SecuredJaxRsEndpoint;
+import org.jboss.eap.qe.ts.common.docker.Docker;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class KeycloakIntegrationHighLevelScenarioTest {
+
+    private static final String KEYCLOAK_REALM_NAME = "foobar";
+
+    private static final String DEFAULT_SERVER_BIND_ADDRESS = "localhost";
+
+    private static final String KEYCLOAK_CONTAINER_NAME = KeycloakIntegrationHighLevelScenarioTest.class.getSimpleName()
+            + "-container";
+
+    private static final int KEYCLOAK_EXPOSED_HTTP_PORT = 8179;
+    private static final int KEYCLOAK_EXPOSED_TOKEN_PORT = 8149;
+    private static final int KEYCLOAK_EXPOSED_MANAGEMENT_PORT = 9988;
+
+    public static Docker keycloakContainer = new Docker.Builder(KEYCLOAK_CONTAINER_NAME,
+            "registry.hub.docker.com/jboss/keycloak:8.0.1")
+                    .setContainerReadyTimeout(2, TimeUnit.MINUTES)
+                    .setContainerReadyCondition(() -> isContainerReady(KEYCLOAK_EXPOSED_HTTP_PORT))
+                    .withPortMapping(KEYCLOAK_EXPOSED_HTTP_PORT + ":8080")
+                    .withPortMapping(KEYCLOAK_EXPOSED_TOKEN_PORT + ":8085")
+                    .withPortMapping(KEYCLOAK_EXPOSED_MANAGEMENT_PORT + ":9990")
+                    .withEnvVar("KEYCLOAK_USER", "admin")
+                    .withEnvVar("KEYCLOAK_PASSWORD", "admin")
+                    .withCmdArg("-b=0.0.0.0")
+                    .withCmdArg("-bmanagement=0.0.0.0")
+                    .build();
+
+    public static KeycloakConfigurator keycloakConfigurator = new KeycloakConfigurator.Builder(KEYCLOAK_REALM_NAME)
+            .adminPassword("admin")
+            .adminUsername("admin")
```

Comment:
"admin" ...could it be moved to constant?

Reply:
Thanks, moving